include/fixed_containers/recursive_reflection.hpp

```diff
@@ -0,0 +1,373 @@
+
+    while (end != std::string_view::npos)
+    {
+        path_components.emplace_back(path_name_chain_string.substr(start, end - start));
+        start = end + 1;
+        end = path_name_chain_string.find(recursive_reflection_detail::PATH_DELIMITER, start);
+    }
+
+    if (start < path_name_chain_string.length())
+    {
+        path_components.emplace_back(path_name_chain_string.substr(start));
+    }
+
+    return PathNameChain(path_components.begin(), path_components.end());
+}
+
+inline std::string path_to_string(const PathNameChain& path_name_chain)
+{
+    if (path_name_chain.empty())
+    {
+        return {};
+    }
+
+    std::string result;
+    bool first = true;
+
+    for (const auto& component : path_name_chain)
+    {
+        if (!first)
+        {
+            result += recursive_reflection_detail::PATH_DELIMITER;
+        }
+        result += component;
+        first = false;
+    }
+
+    return result;
+}
+
+template <typename T>
+concept StrategyCovered =
+    recursive_reflection_detail::ReflectionHandler<std::decay_t<T>>::REFLECTABLE;
+
+/*
+ * This function recursively reflects into a type. (for each path)
+ * The function calls corresponding handler to decide how to recursively reflect into the type.
+ *
+ * See the test for an example of how to do provide custom reflection strategy
+ */
+template <typename S, typename PreFunction, typename PostFunction>
+constexpr void for_each_path_dfs_helper(S&& reflected_object,
+                                        PreFunction&& pre_fn,
+                                        PostFunction&& post_fn,
+                                        in_out<PathNameChain> chain)
+{
+    using Handler = recursive_reflection_detail::ReflectionHandler<std::decay_t<S>>;
+    if constexpr (Handler::REFLECTABLE)
+    {
+        Handler::reflect_into(std::forward<S>(reflected_object),
+                              std::forward<PreFunction>(pre_fn),
+                              std::forward<PostFunction>(post_fn),
+                              in_out{*chain});
+    }
+}
+
+template <StrategyCovered S, typename PreFunction, typename PostFunction>
+constexpr void for_each_path_dfs(S&& reflected_object, PreFunction&& pre_fn, PostFunction&& post_fn)
+{
+    PathNameChain chain = {};
+    for_each_path_dfs_helper(std::forward<S>(reflected_object),
+                             std::forward<PreFunction>(pre_fn),
+                             std::forward<PostFunction>(post_fn),
+                             in_out{chain});
+}
+
+template <StrategyCovered S, typename PreFunction>
+constexpr void for_each_path_dfs(S&& reflected_object, PreFunction&& pre_fn)
+{
+    PathNameChain chain = {};
+    auto dummy_function = [](const auto&, auto&) {};
+    for_each_path_dfs_helper(std::forward<S>(reflected_object),
+                             std::forward<PreFunction>(pre_fn),
+                             dummy_function,
+                             in_out{chain});
+}
+
```

Comment:
Let's add constexpr tests for this file.

Reply:
added tests to use `for_each_path` and all tests are constexpr